src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -174,39 +174,50 @@ private void handleCopy(String detailsToCopy) {
         clipboard.setContent(content);
     }
 
-    /**
-     * Executes the command and returns the result.
-     *
-     * @see seedu.address.logic.Logic#execute(String)
-     */
-    private String executeCommand(String commandText) throws CommandException, ParseException, StorageException {
-        String commandResult;
-        try {
-            commandResult = logic.execute(commandText);
-            logger.info("Result: " + commandResult);
-            showMessage(commandResult);
-        } catch (CommandException | ParseException | StorageException e) {
-            logger.info("An error occurred while executing command: " + commandText);
-            showMessage(e.getMessage());
-            throw e;
-        }
+    public void showMessage(String msg) {
+        resultDisplay.setFeedbackToUser(msg);
+    }
+
+    private class CommandExecutorInator implements CommandExecutor {
```

Comment:
Should this be in a separate file instead? Also, should it be called RecordedCommandExecutor instead?

Reply:
Placing it in a separate file won't work as it depends on attributes from `MainWindow`.